src/controllers/serviceaccounts.go

```diff
@@ -145,6 +147,11 @@ func (c *serviceAccountController) CreateServiceAccount(name, namespace string)
 	if err != nil {
 		return types.ServiceAccount{}, customerrors.NewAPIError(fmt.Sprintf(ErrCouldNotCreateServiceAccount, name, namespace), err)
 	}
+
+	err = CreateTokenRequestSecret(c.ctx, c.client, name, namespace)
+	if err != nil {
+		return types.ServiceAccount{}, err
+	}
```

Comment:
I think a more logical flow would be to only create the secret when a token is requested so that it's not part of the `serviceAccount` controller at all. Instead, when we create the Token, we should check if the secret exists and if it doesn't then we should create it